packages/cli/src/utils.ts

```diff
@@ -153,6 +154,146 @@ export const validFieldsFromPkg = {
     }
     return obj;
   },
+  exports(
+    pkg: Package,
+    hasModuleBuild: boolean,
+    hasBrowserField: boolean,
+    hasWorkerField: boolean,
+    entrypointName: string,
+    forceStrategy?: DistFilenameStrategy
+  ) {
+    let obj: Record<string, ExportsItem> = {
+      ".": exportsHelpers.root(
```

Comment:
i believe that we should also always add `package.json` here - otherwise it might not be possible to do stuff like `require('a/package.json')`

Reply:
I thought about this. The only reason I didn't add it explicitly is that this could be true of any config file and there is no way we could support all the cases. 

Instead, this implementation allows you to manually add those one-off values without triggering an invalid field error. That said, if you ever run `preconstruct fix` you'll likely loose those extra values that you had added manually. 

If you think package.json is common enough, that I'm happy to add that one in.

Reply:
> That said, if you ever run preconstruct fix you'll likely loose those extra values that you had added manually.

This, I think that preconstruct should be in full control of the exports field. If one needs to extend it in any way then we should allow specifying extensions for that in the options for preconstruct (`package.json#preconstruct`). This would avoid accidental removals when running fix etc.

Reply:
package.json has been added as well as a generic `./`: `./` field that I see in many packages.

Reply:
Hm, I wonder if `./` is even valid 🤔 I don't really see any mention of that in the npm docs [here](https://nodejs.org/api/packages.html#package-entry-points) and webpack docs call this out as intentionally ignored (just below the table [here](https://webpack.js.org/guides/package-exports/#support)).

The `./*` is definitely valid though and it can be used to "disable package encapsulation". I believe this is, in general, a bad thing to do - unless somebody really, really needs this, for some specific reason. It could be used as a stop-gap solution though - with the intention of being removed in the future (or with the intention of the default being switched). OTOH - this whole thing will, most likely, have to be hidden behind a feature flag as adding `package.json#exports` might be breaking for package consumers. So we want to make it intentional, to include this, on the package authors' side of things. So if this will be an opt-in anyway - I don't see a reason why we'd want to include this right away. I would be in favor of releasing this without it and gathering feedback from the users - to see if this is actually even needed in practice.

Reply:
That sounds reasonable. I'll remove it.

While we're on the topic, packages like react import custom JSON files like `build.json`. How do you think a lib maintainer could opt into something like that?

Reply:
> While we're on the topic, packages like react import custom JSON files like build.json. How do you think a lib maintainer could opt into something like that?

I've touched the subject in [this comment](https://github.com/preconstruct/preconstruct/pull/435#issuecomment-994144223)